### PR TITLE
HHH-14503 - Migration of tests from jpa/test to orm/test/jpa

### PR DESCRIPTION
**Pull request by jrenaat**
Signed-off-by: Jan Schatteman <jschatte@redhat.com>

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/components/ComponentInWhereClauseTest.java [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+ */
+@TestForIssue(jiraKey = "HHH-6562")
+@Jpa(annotatedClasses = {
+		ComponentInWhereClauseTest.Employee.class,
+		ComponentInWhereClauseTest.Project.class,
+		ComponentInWhereClauseTest.Person.class
+})
+public class ComponentInWhereClauseTest {
+	private Projects projects;
+
```
Comment (jrenaat) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/components/ComponentInWhereClauseTest.java:52]: I did not add a tearDown method, I could get it to work ... (couldn't get past the FK violation exceptions)

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metamodel/MappedSuperclassType2Test.java [hunk 2]
```diff
@@ -17,20 +17,19 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 /**
  * Specifically see if we can access a MappedSuperclass via Metamodel that is not part of a entity hierarchy
  *
  * @author Steve Ebersole
  */
-public class MappedSuperclassType2Test extends BaseUnitTestCase {
+	public class MappedSuperclassType2Test {
```
Comment (dreab8) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/metamodel/MappedSuperclassType2Test.java:32]: I would add `@BaseUnitTest`

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/paths/SingularAttributeJoinTest.java [hunk 3]
```diff
@@ -4,59 +4,38 @@
...
+        integrationSettings = { @Setting(name = AvailableSettings.JPA_METAMODEL_POPULATION, value = "enabled") },
+        xmlMappings = { "org/hibernate/orm/test/jpa/criteria/paths/PolicyAndDistribution.hbm.xml" }
+)
+public class SingularAttributeJoinTest {
 
     @Test
-    public void testEntityModeMapJoinCriteriaQuery() throws Exception {
-        final EntityManager entityManager = entityManagerFactory().createEntityManager();
+    public void testEntityModeMapJoinCriteriaQuery(EntityManagerFactoryScope scope) {
+        final EntityManager entityManager = scope.getEntityManagerFactory().createEntityManager();
```
Comment (dreab8) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/paths/SingularAttributeJoinTest.java:35]: I would use
```
scope.inEntityManager(
				entityManager -> {
```

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/TupleQueryTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+		scope.inTransaction(
+				entityManager -> {
+					entityManager.createQuery( "delete from User" ).executeUpdate();
+				}
+		);
+	}
+
+	@Test
+	public void testGetAliasReturnNullIfNoAliasExist(EntityManagerFactoryScope scope) {
+		scope.inEntityManager(
```
Comment (dreab8) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/TupleQueryTest.java:58]: I would use `inTransaction` and remove the try/catch block

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/TupleQueryTest.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+						TypedQuery<Tuple> query = entityManager.createQuery( "SELECT u.firstName as fn from User u", Tuple.class );
+
+						List<Tuple> result = query.getResultList();
+						List<TupleElement<?>> elements = result.get( 0 ).getElements();
+
+						assertThat( elements.size(), is( 1 ) );
+						final String alias = elements.get( 0 ).getAlias();
+						assertThat( alias, is( "fn" ) );
+					}
+					catch (Exception e) {
```
Comment (dreab8) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/TupleQueryTest.java:94]: the try/catch block is useless here

**Pull request closed without merging**